server/common/corpora.py

```diff
@@ -0,0 +1,81 @@
+"""
+Corpora schema conventions support.  Helper functions for reading.
+
+https://github.com/chanzuckerberg/corpora-data-portal/blob/main/backend/schema/corpora_schema.md
+
+https://github.com/chanzuckerberg/corpora-data-portal/blob/main/backend/schema/corpora_schema_h5ad_implementation.md
+"""
+import json
+
+from server.cli.upgrade import validate_version_str
+
+
+def corpora_get_versions_from_anndata(adata):
+    """
+    Given an AnnData object, return:
+        * None - if not a Corpora object
+        * [ corpora_schema_version, corpora_encoding_version ] - if a Corpora object
+
+    Implements the identification protocol defined in the spect.
+    """
+
+    # per Corpora AnnData spec, this is a corpora file if the following is true
+    if "version" not in adata.uns or "corpora_schema_version" not in adata.uns["version"]:
+        # oops, not corpora
+        return None
+
+    corpora_schema_version = adata.uns["version"].get("corpora_schema_version", None)
+    corpora_encoding_version = adata.uns["version"].get("corpora_encoding_version", None)
+
+    # TODO: spec says these must be SEMVER values, so check.
+    if validate_version_str(corpora_schema_version) and validate_version_str(corpora_encoding_version):
+        return [corpora_schema_version, corpora_encoding_version]
+
+    return None
+
+
+def corpora_get_props_from_anndata(adata):
+    """
+    Get Corpora dataset properties from an AnnData
+    """
+    versions = corpora_get_versions_from_anndata(adata)
+    if versions is None:
+        return None
+    [corpora_schema_version, corpora_encoding_version] = versions
+    version_is_supported = (
+        corpora_schema_version is not None
+        and corpora_encoding_version is not None
+        and corpora_schema_version.startswith("1.")
```

Comment:
Can we pull out the supported schema versions into a utility file or some sort of constants file. I see that `cxgtool.py` also checks the versions so it would be helpful to have the supported versions in a single location that can get updated instead of having to remember a couple places to check.

Reply:
Can you expand on what problem this solves?  I bias away from overly complicated config & code structure unless there is a clear value.   Right now, this file is entirely self-contained wrt semantics and readability.  Given how simple the code is, I'm not sure splitting helps.

LMK.

Reply:
I figured that because this particular check is written word-for-word twice (lines 551-556) in a completely different file `cxgtool.py`, it seems preferable to consolidate into a single location. Otherwise you run into the issue of having to remember to update in both locations or run into bugs.

Reply:
Ah, I see - you think both files will always have the same version check.  I was not assuming that.  But for now, I'll DRY it.